### PR TITLE
feat: add support for requesting "Specials" for TV Shows

### LINKED ISSUE #779: Support for requesting TV show specials
**Is your feature request related to a problem? Please describe.**
Currently, you can't request a special episode in Overseerr.
Is it possible to make an option available to request specials?

**Describe the solution you'd like**
Make it possible to the end-user to create an option to request specials.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by AhmedNSidd**
#### Description

This pull request is responsible for adding support in Overseerr for requesting "Special" episodes for TV Shows. This request has become especially pertinent when you consider shows like "Doctor Who". These shows have Specials that are critical to understanding the plot of a TV show.

##### The TV Show Page
If you navigate to any show with Specials, you'll see the "Specials" season just before "Season 1". 
**NOTE**: In this example, Doctor Who has 13 Seasons + Specials. However, in the header for the TV show, we STILL say that the show has 13 seasons (we don't include specials as a "Season" in the headers).

CleanShot 2023-12-24 at 14 56 10@2x

##### The TV Show Page (Request Modal)

**NOTE**: Instead of having the table say `Season 0` which it would normally say for the Specials season, it says "Specials". Whether it should say `Season 0` or `Specials` is a discussion that could be worth having for this PR, but I personally went with `Specials` for more user-friendliness.

CleanShot 2023-12-24 at 15 01 51@2x

##### The TV Show Page (Manage Series Sidebar)

**NOTE**: Instead of having the Badge say `0` which it would normally say for the Specials season, it says "Specials". Whether it should say `0` or `Specials` is a discussion that could be worth having for this PR, but I personally went with `Specials` for more user-friendliness.

CleanShot 2023-12-24 at 14 59 36@2x

##### The Requests Page

CleanShot 2023-12-24 at 15 02 50@2x

#### To-Dos

- [x] Successful build `yarn build`
- [X] Translation keys `yarn i18n:extract`
- [ ] Database migration (if required)

#### Issues Fixed or Closed

- Fixes #779

### CODE REVIEW

**Comment by sct** [comment]
Can we remove the `yarn.lock` from this PR? It shouldn't be getting updated here.

File: src/components/RequestModal/TvRequestModal.tsx
Comment (westdom) [inline src/components/RequestModal/TvRequestModal.tsx:635]: If the season number is 1 won't we still need `intl.formatMessage(messages.extras)`?

File: src/components/TvDetails/index.tsx
Comment (westdom) [inline src/components/TvDetails/index.tsx:206]: I might be wrong but can't we simplify this to:
```suggestion
    (season) => season.episodeCount !== 0
```
Then we have no need for `showHasSpecials`?

File: src/components/RequestModal/TvRequestModal.tsx
Comment (AhmedNSidd) [reply src/components/RequestModal/TvRequestModal.tsx:635]: I'm not sure I understand @westdom 

- Firstly, I refactored the variable from "Extras" to "Specials" because typically, Season 0 of a TV Show is typically referred to as "Specials". So by the change I made in line 44 of this file, it deprecates the usage of Extras entirely.
- Secondly, if the season number is 1 in this ternary operator, then it would use the code in line 636 of this file; specifically the message `messages.seasonnumber`. So, Extras would not play a role there either.

File: src/components/TvDetails/index.tsx
Comment (AhmedNSidd) [reply src/components/TvDetails/index.tsx:206]: Good spot @westdom . Yes, you are right, and that's what I did initially, but I made the intentional decision of ensuring that the `seasonCount` variable does not include TV Show Specials.

Why? Because in the header / banner of every TV Show, it is written how many seasons are in the TV Show in question. I made the intentional decision that, for example, if a TV Show like `Doctor Who` has 13 Seasons + Specials, then we wouldn't say `Doctor Who` has 14 Seasons, as for the average user, they don't really consider specials as a separate season. Hence, it was important that `seasonCount` did not include TV Show Specials in its count.

Because of this, the `showHasSpecials` variable was needed.

**Review by AhmedNSidd — COMMENTED** [review]
@sct I made a commit that was responsible for undoing `yarn.lock` update. Thanks for letting me know.

@westdom Thank you for reviewing the PR. I responded to your suggestions, but made no code changes. Let me know if you have any more thoughts. Otherwise, I believe the conversations can be marked as resolved.

File: src/components/TvDetails/index.tsx
Comment (westdom) [reply src/components/TvDetails/index.tsx:206]: Perfect that makes sense now, thanks!

**Comment by Panda260** [comment]
Any updates on this?

**Comment by AhmedNSidd** [comment]
@Panda260 No new updates.. I've updated the branch to be rebased on top of the `develop` branch. I'm not sure what the next steps are for having this merged in. I believe I need an approval from @sct / @TheCatLady / @danshilm / @OwsleyJr

**Comment by OwsleyJr** [comment]
> @Panda260 No new updates.. I've updated the branch to be rebased on top of the `develop` branch. I'm not sure what the next steps are for having this merged in. I believe I need an approval from @sct / @TheCatLady / @danshilm / @OwsleyJr

Would need to resolve the pending comments and then we can get this merged.

**Comment by Lockszmith-GH** [comment]
Awsome features, but just so I'm clear - this is an all-or-nothing request for specials, and does not allow individual special episode report - right?

**Comment by BlackMamba248** [comment]
Is there any update for this?

**Comment by AhmedNSidd** [comment]
> Would need to resolve the pending comments and then we can get this merged.

@OwsleyJr Can you please clarify what conversations are yet to be resolved? As far as I can see, all conversations have been resolved. Anyway, I've also taken the time now to rebase from `upstream/main` again. CC: @Panda260 

-----

> Awsome features, but just so I'm clear - this is an all-or-nothing request for specials, and does not allow individual special episode report - right?

@Lockszmith-GH It's been a second since I've worked on this, but yes, as far as I remember, this is an all-or-nothing request. The entire season will be marked in Sonarr as monitored, and Sonarr will then make the effort of searching through for every special episode.

------------

> Is there any update for this?

@BlackMamba248 Currently waiting for a review from Code Owners... Wait and see for now.

**Comment by OwsleyJr** [comment]
> > Would need to resolve the pending comments and then we can get this merged.
> 
> 
> 
> @OwsleyJr Can you please clarify what conversations are yet to be resolved? As far as I can see, all conversations have been resolved. Anyway, I've also taken the time now to rebase from `upstream/main` again.
> 
> 
> 
> -----
> 
> 
> 
> > Awsome features, but just so I'm clear - this is an all-or-nothing request for specials, and does not allow individual special episode report - right?
> 
> 
> 
> @Lockszmith-GH It's been a second since I've worked on this, but yes, as far as I remember, this is an all-or-nothing request. The entire season will be marked in Sonarr as monitored, and Sonarr will then make the effort of searching through for every special episode.
> 
> 
> 
> ------------
> 
> 
> 
> > Is there any update for this?
> 
> 
> 
> @BlackMamba248 Currently waiting for a review from Code Owners... Wait and see for now.
> 
> 
> 
> 

I never submitted my review. That's my fault

File: src/components/RequestBlock/index.tsx
Comment (AhmedNSidd) [reply src/components/RequestBlock/index.tsx]: Done! Thanks for pointing it out.

File: src/i18n/locale/en.json
Comment (AhmedNSidd) [reply src/i18n/locale/en.json]: Done!

**Review by AhmedNSidd — COMMENTED** [review]
> I never submitted my review. That's my fault

@OwsleyJr All good. Thank you for the review! I've addressed your comments and marked the relevant conversations as resolved. Please take a look at my changes to validate whether the concerns have been addressed or not.

**Review by OwsleyJr — APPROVED** [review]
LGTM!

**Comment by OwsleyJr** [comment]
@all-contributors please add @AhmedNSidd for code

**Comment by BlackMamba248** [comment]
> > Would need to resolve the pending comments and then we can get this merged.
> 
> @OwsleyJr I believe all conversations are marked as resolved now. I've also taken the time now to rebase from `upstream/main` again.
> 
> > Awsome features, but just so I'm clear - this is an all-or-nothing request for specials, and does not allow individual special episode report - right?
> 
> @Lockszmith-GH It's been a second since I've worked on this, but yes, as far as I remember, this is an all-or-nothing request. The entire season will be marked in Sonarr as monitored, and Sonarr will then make the effort of searching through for every special episode.
> 
> > Is there any update for this?
> 
> @BlackMamba248 Currently waiting for a review from Code Owners... Wait and see for now.


Thanks for the update! I've been trying it out and it seems to be inconsistent for the specials area for anime. Mainly It does not always appear to show the specials from TVDB that are listed as movies or season recaps or other naming schemes under the specials area. Is there a way to incorporate that to show everything in specials or at least include the related anime movies in this case as they are generally listed as specials but sometimes categorized differently under there

**Pull request merged**
